crates/dids/src/method/spruce_mappers/document.rs

```diff
@@ -0,0 +1,277 @@
+                            serde_json::to_string(&obj).unwrap_or_default()
+                        }
+                    })
+                    .collect(),
+            ),
+        };
+
+        let verification_methods = spruce_document
+            .verification_method
+            .unwrap_or_default()
+            .into_iter()
+            .map(VerificationMethod::from_spruce)
+            .collect::<Result<Vec<_>, String>>()?;
+
+        let authentication = spruce_document.authentication.map(|vms| {
+            vms.into_iter()
+                .map(|vm| vm.get_id(&spruce_document.id))
+                .collect::<Vec<_>>()
+        });
+
+        let assertion_method = spruce_document.assertion_method.map(|vms| {
+            vms.into_iter()
+                .map(|vm| vm.get_id(&spruce_document.id))
+                .collect::<Vec<_>>()
+        });
+
+        let key_agreement = spruce_document.key_agreement.map(|vms| {
+            vms.into_iter()
+                .map(|vm| vm.get_id(&spruce_document.id))
+                .collect::<Vec<_>>()
+        });
+
+        let capability_invocation = spruce_document.capability_invocation.map(|vms| {
+            vms.into_iter()
+                .map(|vm| vm.get_id(&spruce_document.id))
+                .collect::<Vec<_>>()
+        });
+
+        let capability_delegation = spruce_document.capability_delegation.map(|vms| {
+            vms.into_iter()
+                .map(|vm| vm.get_id(&spruce_document.id))
+                .collect::<Vec<_>>()
+        });
+
+        let service = spruce_document
+            .service
+            .map(|services| {
+                services
+                    .into_iter()
+                    .map(Service::from_spruce)
+                    .collect::<Result<Vec<_>, String>>()
+            })
+            .transpose()?;
+
+        Ok(Document {
+            id: spruce_document.id,
+            context,
+            controller: spruce_document
+                .controller
+                .map(|c| c.into_iter().map(|did| did.to_string()).collect()),
+            also_known_as: spruce_document
+                .also_known_as
+                .map(|aka| aka.into_iter().map(|uri| uri.to_string()).collect()),
+            verification_method: verification_methods,
+            authentication,
+            assertion_method,
+            key_agreement,
+            capability_invocation,
+            capability_delegation,
+            service,
+        })
+    }
+}
+
+impl VerificationMethod {
+    pub fn from_spruce(
+        spruce_verification_method: SpruceVerificationMethod,
+    ) -> Result<Self, String> {
+        match spruce_verification_method {
+            SpruceVerificationMethod::Map(ssi_vmm) => {
+                let spruce_jwk = ssi_vmm.get_jwk()?;
+                let alg = match spruce_jwk.algorithm {
+                    Some(Algorithm::ES256K) => "ES256K",
+                    Some(Algorithm::EdDSA) => "EdDSA",
```

Comment:
should switch out EdDSA to Ed25519 here and elsewhere
https://www.ietf.org/archive/id/draft-ietf-jose-fully-specified-algorithms-02.html

Reply:
Great callout. I did this but it'll be better to dedicate the time to do this right, so created a ticket and will follow up with it https://github.com/TBD54566975/web5-rs/issues/128